synapse/lib/cell.py

```diff
@@ -958,6 +972,9 @@ async def __anit__(self, dirn, conf=None, readonly=False, parent=None):
         self.onfini(root.fini)
         root.onfini(self.fini)
 
+        # Now we can register our own finis
+        self.onfini(_onFiniCellGuid)
```

Comment:
One thought, this may be worth *guaranteeing* it gets run last in the fini order by over-riding the fini() method with something like
```
async def fini(self):
    retn = await Cell.fini(self)
    await self._onFiniCellGuid()
    return retn
```